File: Reference Resources/rosnode_skeleton.py
#!/usr/bin/env python

import rospy
import sys
from std_msgs.msg import String
from mill_controller.msg import Trajectory


# Use this publisher to send the output
pub = rospy.Publisher('trajectory_input',Trajectory,queue_size=1)



def callback(input):
    # What does the callback do?





def main():
    # Initialize the ROS node and Subscriber
    rospy.init_node("text_input_parser_node")
    rospy.Subscriber('string_input',String,callback)

    # Spin until shut down
    rospy.spin()





if __name__ == '__main__':
    main()

File: Reference Resources/rosnode_skeleton.cpp
// Includes
#include<ros/ros.h>
#include<sensor_msgs/PointCloud2.h>
#include<sensor_msgs/Image.h>

// Declare publisher
ros::Publisher pub;


void callback(const sensor_msgs::Image &msg)
{
    printf("call");
}


int main(int argc, char **argv)
{
    // Initialize node and declare NodeHandle
    ros::init(argc,argv,"node_name");
    ros::NodeHandle nh;

    // Subscribe to topic with 1 message buffer
    ros::Subscriber sub = nh.subscribe("input_topic", 1, callback);

    // Announce publisher with 1 message buffer
    pub = nh.advertise<sensor_msgs::PointCloud2>("output_topic", 1);

    // Spin until shutdown
    ros::spin();

    return 0;
}
